zkevm-circuits/src/evm_circuit/execution/return_revert.rs

```diff
@@ -155,7 +158,7 @@ impl<F: Field> ExecutionGadget<F> for ReturnRevertGadget<F> {
                         + not::expr(is_success.expr())
                             * cb.curr.state.reversible_write_counter.expr(),
                 ),
-                gas_left: Delta(-memory_expansion.gas_cost()),
+                gas_left: Delta(-memory_expansion.gas_cost() - code_deposit_cost.expr()),
```

Comment:
Is it ok to subtract it when returning/reverting? Wouldn't we need to do this in the create operation as there might be other opcodes after the create operation and before the return operation (and these operations might end up in OOG because of the gas subtracted for byte code storage in the create operation)?

Reply:
As I understand, we can't get deployment bytecode in CREATE/CREATE2 context and It's only available in RETURN context. The steps would be like 1. CREATE/CREATE2 creates the context, 2. run the init code, 3. part of init code deployed as contract bytecode and then 4. call RETURN.

After Init bytecode is executed and deploys contract code, we can get the contract bytecode. That's why we calculate `code_deposit_cost` in RETURN.

Reply:
Yes, I see it now, I misunderstood how contract creation is complied to byte codes ([this article ](https://www.rareskills.io/post/ethereum-contract-creation-code)explains it well). Thanks!